cli/assets/templates/zig/build.zig

```diff
@@ -14,7 +25,12 @@ test "stack version check" {
     try expect(!try version_supports_stack_first(try parse("0.8.0")));
 
     try expect(!try version_supports_stack_first(try parse("0.9.0")));
-    try expect(!try version_supports_stack_first(try parse("0.9.1")));
+    try expect(!try version_supports_stack_first(try parse("0.9.1-dev.259")));
+    try expect(try version_supports_stack_first(try parse("0.9.1")));
+
+    // Conservatively don't recognize tags other than 'dev'.
+    try expect(!try version_supports_stack_first(try parse("0.10.0-aev.259")));
+    try expect(!try version_supports_stack_first(try parse("0.10.0-zev.259")));
```

Comment:
I think we should fail on exotic zig versions, then the user would handle this code themselves to fix it. We could add a recommendation on the website to download nightly or stable 0.9.1. wdyt?

Reply:
I slightly prefer the current behavior of just falling back to safe legacy behavior and printing a warning. No need to force them into a session of debugging our build script to be able to build at all.

Reply:
> We could add a recommendation on the website to download nightly or stable 0.9.1

Probably not worth it IMO. It's another place we'd need to keep a version number up to date. "For best results use a recent version" mostly goes without saying :)

Reply:
Well then, the descision is behind @aduros